src/components/Card.js

```diff
@@ -1,22 +1,39 @@
 export default class Card {
-    constructor(cardData, cardSelector, handleCardClick) {
-        this._name = cardData.title;
+    constructor(cardData, cardSelector, handleCardClick, handleLikeClick, handleDeleteClick) {
+        this._name = cardData.name;
         this._link = cardData.link;
+        this._id = cardData._id;
+        this._isLiked = cardData.isLiked;
         this._cardSelector = cardSelector;
         this._handleCardClick = handleCardClick;
+        this._handleLikeClick = handleLikeClick;
+        this._handleDeleteClick = handleDeleteClick;
     }
 
     _setEventListeners() {
-        this._element.querySelector(".card__like-button").addEventListener('mousedown', () => this._handleLikeButtonClick());
-        this._element.querySelector(".card__delete-button").addEventListener('mousedown', () => this._handleDeleteCard());
+        this._element.querySelector(".card__like-button").addEventListener('mousedown', () => this._handleLikeClick(this));
+        this._element.querySelector(".card__delete-button").addEventListener('mousedown', () => this._handleDeleteClick(this));
         this._element.querySelector(".card__image").addEventListener('mousedown', () => this._handlePreviewPicture());
     }
 
-    _handleLikeButtonClick() {
-        this._element.querySelector(".card__like-button").classList.toggle("card__like-button_active");
+    _handleLikeButton() {
+         if (this._isLiked) {
+            this._element.querySelector(".card__like-button").classList.add("card__like-button_active")
```

Comment:
`this._element.querySelector(".card__like-button")`

**Could be improved**

All elements that are found every time you call a method should be defined as **class fields** in the constructor, `_setEventListeners` (or in `generateCard`) only **1 time** not to waste resources on searching them again and again.

`Class fields` (the so called **class variables** with `this`) are accessible inside any method of the class. You shouldn’t pass **class fields** into the call of any method because you can use class fields inside all methods.
There can be `this._inputs`, `this._submitButton`, `this._cardImage`, `this._form` , `this._likeButton` and so on.

```js
  getView() {
    this._element = this._getTemplate();
    this._likeButton = this._element.querySelector(".card__like-button");
    this._cardImage = this._element.querySelector(".card__image");
    this._trashButton = this._element.querySelector(".card__trash-button");

```